pyproject.toml

```diff
@@ -40,6 +40,7 @@ dependencies = [
   "ragas~=0.2.14",
   "rich~=13.9",
   "uvicorn[standard]~=0.32.0",
+  "weave>=0.51.43",
```

Comment:
I have added weave as a dependency for now so that testing this PR would be easier. 

If this is okay, we can leave it as it is. The reason I made this a requirement is because of changes made to `async_otel_listner.py` file.

Reply:
One option we have is to create a new sub-package which would be `agentiq-weave` to contain all of the weave integration pieces. It would only be this singular plugin for now but it would solve the dependency issue.

Generally, we are trying to keep the core library dependencies light. What do you think of this option?